docs/user_docs/guides/abstraction.md

```diff
@@ -0,0 +1,114 @@
+---
+title: "Abstraction"
+sidebar_position: 3
+---
+
+## What is Abstraction
+
+Abstraction refers to a simplified representation of an entity, typically used in computing. It allows for the concealment of specific details while presenting the most relevant information to the programmer. Each abstraction is tailored to suit a specific need, and can greatly enhance the usability of a given entity. In the context of KCL, abstraction can make code easier to understand and maintain, while also simplifying the user interface. 
+
+It should be noted that code abstraction is not meant to reduce code size, but rather to improve maintainability and extendability. During the process of abstracting code, factors such as reusability, readability, and scalability should be taken into consideration, and the code should be optimized as needed.
+
+## Use KCL for Abstraction
```

Comment:
Add some pre-knowledge and refs about Docker and Kubernetes

Reply:
> Add some pre-knowledge and refs about Docker and Kubernetes

fixed.